vsphere_plugin_common/__init__.py

```diff
@@ -1058,6 +1118,10 @@ def create_server(self,
             vm_memory=memory,
         )
 
+        # Correct the network name for all networks from relationships
+        for network in networks:
+            network['name'] = self._get_connected_network_name(network)
```

Comment:
If you did this block before `_validate_inputs()` you'd halve the number of calls to `_get_connected_network_name()`. Am I missing an obvious reason not to do that?